crates/driver/src/infra/solver/mod.rs

```diff
@@ -187,7 +186,15 @@ impl Solver {
     ) {
         let body =
             serde_json::to_string(&dto::Notification::new(auction_id, solution_id, kind)).unwrap();
-        let url = shared::url::join(&self.config.endpoint, "notify");
+
+        let url = match shared::url::replace_endpoint(self.config.endpoint.clone(), "notify") {
```

Comment:
Nit: I'd be in favour of making this endpoint configurable as well (and even opt-in as we are currently getting a lot of 404 logs of solvers that are not implementing this)